src/polly/src/event_manager.rs

```diff
@@ -202,21 +202,14 @@ impl EventManager {
         Ok(())
     }
 
-    /// Register a new event handler.
-    /// handler.init() will specify the pollable and event set.
-    ///
     /// Returns the handler wrapped in an Arc<Mutex>.
-    pub fn register<T: EventHandler + 'static>(&mut self, handler: T) -> Result<Arc<Mutex<T>>> {
-        let wrapped_handler = Arc::new(Mutex::new(handler));
-        self.register_protected(wrapped_handler.clone())?;
-        Ok(wrapped_handler)
+    pub fn wrap_handler<T: EventHandler + 'static>(handler: T) -> Arc<Mutex<T>> {
```

Comment:
Since this only appears to call `Arc::new(Mutex::new(...))`, is it really needed? It looks like invoking that sequence ourselves is also a bit shorter and clearer than writing `EventManager::wrap_handler(...)`.

Reply:
done